lib/mlir/Dialect/Clift/Transforms/CBackend.cpp

```diff
@@ -93,6 +93,7 @@ struct EmitCPass : clift::impl::CliftEmitCBase<EmitCPass> {
 
     llvm::raw_null_ostream NullStream;
     ptml::CTypeBuilder B(NullStream, *Model, ptml::CBuilder(Tagless));
+    B.collectInlinableTypes();
```

Comment:
Ok, but you're not passing the `ConfigurationOptions` to the `CTypeBuilder`.

See what is done [here](https://github.com/revng/revng/blob/f616f30d217a1fc516a6128396a6eb6f187893c5/lib/Backend/DecompilePipe.cpp#L34)

Can you please do that too? Otherwise here the stack inlining will always be disabled.

I know at the moment you're not using it and there's no tests. But we do already have a bunch of separate tests for inlining, even if they don't cover all. And it's worth arranging things so that when we plug this thing in the pipeline everything works as smoothly as possible.

Yon can use [this constructor](https://github.com/revng/revng/blob/f616f30d217a1fc516a6128396a6eb6f187893c5/include/revng/TypeNames/PTMLCTypeBuilder.h#L128) for that.

Reply:
Actually, forget about it.
This is just for the transform used in `clift-opt`. It's fine for that to take the default for the `ConfigurationOptions`. Whenever we make the actual `Pipe`, that will have to properly propagate the `ConfigurationOptions`, but that's not the case here.